File: drivingSchool/BLL/js/component/ajax/message.js
import {messageAjax} from "js/common/variable";
export class Message{
    static insertMessage(sender, content, receiver) {
        let returnValue = false;
        $.ajax({
            url: messageAjax,
            async: false,
            type: "POST",
            data: {
                "key": "insertMessage",
                "sender": sender,
                "content": content,
                "receiver": receiver
            },
            success: result => {
                returnValue = result.toLowerCase() === "true";
            }
        });
        return returnValue;
    }
    static getMessageList(type, name, newMessageNum) {
        let returnValue = false;
        $.ajax({
            url: messageAjax,
            async: false,
            type: "POST",
            data: {
                "key": "getMessageList",
                "type": type,
                "name": name,
                "top": newMessageNum
            },
            success: result => {
                returnValue = result.substr(0, result.length - 1).split(";");
            }
        });
        return returnValue;
    }
    
    static getSentMessage(name){
        let returnValue = false;
        $.ajax({
            url: messageAjax,
            type: "POST",
            async: false,
            data: {
                "key": "getSentMessage",
                "name": name
            },
            success: result => {
                returnValue = result.substr(0, result.length - 1).split(";");
            }
        });
        return returnValue;
    }
}

File: drivingSchool/DAL/Message.cs
using System;
using System.Data.SqlClient;
using drivingSchool.Model;

namespace drivingSchool.DAL {
	public class Message {
		static public string getMessageList (string type, string name, string top) {
			string data = "";
			string command = "";
			if (type == "onlyNew") {
				command = $"select top {Convert.ToInt32(top)} * from message where receiver = '{name}' order by addTime desc";
			} else {
				command = $"select * from message where receiver = '{name}'";
			}
			SqlDataReader read = SQL.getReader (command);
			while (read.Read ()) {
				data += read["sender"] + "=" + read["content"] + "=" + read["addTime"] + ";";
			}
			Coach.initZero (name, "unread");
			SQL.Dispose ();
			return data;
		}
		static public bool insertMessage (MessageModel message) {
			string command = $"insert into message (sender,receiver,content) values ('{message.sender}','{message.receiver}','{message.content}')";
			if (SQL.Excute (command)) {
				Coach.UpdateNewMessage (message.receiver);
				return true;
			}
			return false;
		}
		static public string getSentMessage (string sender) {
			string data = "";
			SqlDataReader read = SQL.getReader ($"select * from message where sender = '{sender}'");
			while (read.Read ()) {
				data += read["receiver"] + "=" + read["content"] + "=" + read["addTime"] + ";";
			}
			return data;
		}
	}
}
